IllegalArgumentException while loading GLTF model

When I try to load [the "CesiumMan" model](https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/CesiumMan/glTF) using [TestGltfLoading](https://github.com/jMonkeyEngine/jmonkeyengine/blob/master/jme3-examples/src/main/java/jme3test/model/TestGltfLoading.java), I get an exception:

```console
Oct 28, 2023 6:55:20 PM com.jme3.system.JmeDesktopSystem initialize
INFO: Running on jMonkeyEngine 3.7.0-SNAPSHOT
 * Branch: master
 * Git Hash: b86310b
 * Build Date: 2023-10-28
Oct 28, 2023 6:55:21 PM com.jme3.system.lwjgl.LwjglContext printContextInitInfo
INFO: LWJGL 2.9.5 context running on thread jME3 Main
 * Graphics Adapter: null
 * Driver Version: null
 * Scaling Factor: 1
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.jme3.util.ReflectionAllocator (file:/home/sgold/NetBeansProjects/jmonkeyengine/jme3-core/build/libs/jme3-core-3.7.0-SNAPSHOT.jar) to method sun.nio.ch.DirectBuffer.cleaner()
WARNING: Please consider reporting this to the maintainers of com.jme3.util.ReflectionAllocator
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
Oct 28, 2023 6:55:21 PM com.jme3.renderer.opengl.GLRenderer loadCapabilitiesCommon
INFO: OpenGL Renderer Information
 * Vendor: NVIDIA Corporation
 * Renderer: GeForce GT 545/PCIe/SSE2
 * OpenGL Version: 3.2.0 NVIDIA 390.157
 * GLSL Version: 1.50 NVIDIA via Cg compiler
 * Profile: Core
Oct 28, 2023 6:55:21 PM com.jme3.audio.openal.ALAudioRenderer initOpenAL
INFO: Audio Renderer Information
 * Device: OpenAL Soft
 * Vendor: OpenAL Community
 * Renderer: OpenAL Soft
 * Version: 1.1 ALSOFT 1.15.1
 * Supported channels: 64
 * ALC extensions: ALC_ENUMERATE_ALL_EXT ALC_ENUMERATION_EXT ALC_EXT_CAPTURE ALC_EXT_DEDICATED ALC_EXT_disconnect ALC_EXT_EFX ALC_EXT_thread_local_context ALC_SOFT_loopback
 * AL extensions: AL_EXT_ALAW AL_EXT_DOUBLE AL_EXT_EXPONENT_DISTANCE AL_EXT_FLOAT32 AL_EXT_IMA4 AL_EXT_LINEAR_DISTANCE AL_EXT_MCFORMATS AL_EXT_MULAW AL_EXT_MULAW_MCFORMATS AL_EXT_OFFSET AL_EXT_source_distance_model AL_LOKI_quadriphonic AL_SOFT_buffer_samples AL_SOFT_buffer_sub_data AL_SOFTX_deferred_updates AL_SOFT_direct_channels AL_SOFT_loop_points AL_SOFT_source_latency
Oct 28, 2023 6:55:21 PM com.jme3.audio.openal.ALAudioRenderer initOpenAL
WARNING: Pausing audio device not supported.
Oct 28, 2023 6:55:21 PM com.jme3.audio.openal.ALAudioRenderer initOpenAL
INFO: Audio effect extension version: 1.0
Oct 28, 2023 6:55:21 PM com.jme3.audio.openal.ALAudioRenderer initOpenAL
INFO: Audio max auxiliary sends: 4
Oct 28, 2023 6:55:21 PM com.jme3.app.LegacyApplication handleError
SEVERE: Uncaught exception thrown in Thread[jME3 Main,5,main]
com.jme3.asset.AssetLoadException: An error occurred loading CesiumMan/glTF/CesiumMan.gltf
	at com.jme3.scene.plugins.gltf.GltfLoader.loadFromStream(GltfLoader.java:181)
	at com.jme3.scene.plugins.gltf.GltfLoader.load(GltfLoader.java:107)
	at com.jme3.asset.DesktopAssetManager.loadLocatedAsset(DesktopAssetManager.java:274)
	at com.jme3.asset.DesktopAssetManager.loadAsset(DesktopAssetManager.java:390)
	at com.jme3.asset.DesktopAssetManager.loadModel(DesktopAssetManager.java:441)
	at jme3test.model.TestGltfLoading.loadModel(TestGltfLoading.java:234)
	at jme3test.model.TestGltfLoading.loadModel(TestGltfLoading.java:229)
	at jme3test.model.TestGltfLoading.simpleInitApp(TestGltfLoading.java:146)
	at com.jme3.app.SimpleApplication.initialize(SimpleApplication.java:240)
	at com.jme3.system.lwjgl.LwjglAbstractDisplay.initInThread(LwjglAbstractDisplay.java:142)
	at com.jme3.system.lwjgl.LwjglAbstractDisplay.run(LwjglAbstractDisplay.java:224)
	at java.base/java.lang.Thread.run(Thread.java:829)
Caused by: java.lang.IllegalArgumentException: Material parameter is not defined: alpha
	at com.jme3.material.Material.checkSetParam(Material.java:474)
	at com.jme3.material.Material.setParam(Material.java:490)
	at com.jme3.scene.plugins.gltf.MaterialAdapter.setParam(MaterialAdapter.java:106)
	at com.jme3.scene.plugins.gltf.GltfLoader.readMaterial(GltfLoader.java:660)
	at com.jme3.scene.plugins.gltf.GltfLoader.readMeshPrimitives(GltfLoader.java:473)
	at com.jme3.scene.plugins.gltf.GltfLoader.readNode(GltfLoader.java:252)
	at com.jme3.scene.plugins.gltf.GltfLoader.readChild(GltfLoader.java:301)
	at com.jme3.scene.plugins.gltf.GltfLoader.readChild(GltfLoader.java:309)
	at com.jme3.scene.plugins.gltf.GltfLoader.readChild(GltfLoader.java:309)
	at com.jme3.scene.plugins.gltf.GltfLoader.readScenes(GltfLoader.java:218)
	at com.jme3.scene.plugins.gltf.GltfLoader.loadFromStream(GltfLoader.java:155)
	... 11 more
```

I have reason to believe this is a recent regression. I'll attempt to pin it down.

--- a/jme3-plugins/src/gltf/java/com/jme3/scene/plugins/gltf/PBRMaterialAdapter.java
+++ b/jme3-plugins/src/gltf/java/com/jme3/scene/plugins/gltf/PBRMaterialAdapter.java
@@ -73,9 +73,11 @@ public abstract class PBRMaterialAdapter extends MaterialAdapter {
                     break;
                 case "BLEND":
                     getMaterial().getAdditionalRenderState().setBlendMode(RenderState.BlendMode.Alpha);
-                    // Alpha is a RenderState not a Material Parameter, so return null
-                    return null;
+                    break;
             }
+
+            // Alpha mode is handled here, so return null
+            return null;
         } else if (param.getName().equals("doubleSided")) {
             boolean doubleSided = (boolean) param.getValue();
             if (doubleSided) {

--- a/jme3-plugins/src/gltf/java/com/jme3/scene/plugins/gltf/PBRMaterialAdapter.java
+++ b/jme3-plugins/src/gltf/java/com/jme3/scene/plugins/gltf/PBRMaterialAdapter.java
@@ -64,34 +64,43 @@ public abstract class PBRMaterialAdapter extends MaterialAdapter {
 
     @Override
     protected MatParam adaptMatParam(MatParam param) {
-        if (param.getName().equals("alpha")) {
-            String alphaMode = (String) param.getValue();
-            switch (alphaMode) {
-                case "MASK":
-                    // "MASK" -> BlendMode.Off
-                    getMaterial().setFloat("AlphaDiscardThreshold", MASK_ALPHA_DISCARD);
-                    break;
-                case "BLEND":
-                    getMaterial().getAdditionalRenderState().setBlendMode(RenderState.BlendMode.Alpha);
-                    // Alpha is a RenderState not a Material Parameter, so return null
-                    return null;
-            }
-        } else if (param.getName().equals("doubleSided")) {
-            boolean doubleSided = (boolean) param.getValue();
-            if (doubleSided) {
-                //Note that this is not completely right as normals on the back side will be in the wrong direction.
-                getMaterial().getAdditionalRenderState().setFaceCullMode(RenderState.FaceCullMode.Off);
-            }
-            // FaceCulling is a RenderState not a Material Parameter, so return null
-            return null;
-        } else if (param.getName().equals("NormalMap")) {
-            //Set the normal map type to OpenGl
-            getMaterial().setFloat("NormalType", 1.0f);
-        } else if (param.getName().equals("LightMap")) {
-            //Gltf only supports AO maps (gray scales and only the r channel must be read)
-            getMaterial().setBoolean("LightMapAsAOMap", true);
-        } else if (param.getName().equals("alphaCutoff")) {
-            getMaterial().setFloat("AlphaDiscardThreshold", (float)param.getValue());
+        switch (param.getName()) {
+            case "alpha":
+                String alphaMode = (String) param.getValue();
+                switch (alphaMode) {
+                    case "MASK":
+                        // "MASK" -> BlendMode.Off
+                        getMaterial().setFloat("AlphaDiscardThreshold", MASK_ALPHA_DISCARD);
+                        break;
+                    case "BLEND":
+                        getMaterial().getAdditionalRenderState().setBlendMode(RenderState.BlendMode.Alpha);
+                        break;
+                }
+
+                // Alpha is a RenderState not a Material Parameter, so return null
+                return null;
+            case "doubleSided":
+                boolean doubleSided = (boolean) param.getValue();
+                if (doubleSided) {
+                    //Note that this is not completely right as normals on the back side will be in the wrong direction.
+                    getMaterial().getAdditionalRenderState().setFaceCullMode(RenderState.FaceCullMode.Off);
+                }
+
+                // FaceCulling is a RenderState not a Material Parameter, so return null
+                return null;
+            case "NormalMap":
+                //Set the normal map type to OpenGl
+                getMaterial().setFloat("NormalType", 1.0f);
+                break;
+            case "LightMap":
+                //Gltf only supports AO maps (gray scales and only the r channel must be read)
+                getMaterial().setBoolean("LightMapAsAOMap", true);
+                break;
+            case "alphaCutoff":
+                getMaterial().setFloat("AlphaDiscardThreshold", (float) param.getValue());
+                break;
+            default:
+                break;
         }
 
         return param;

--- a/jme3-plugins/src/gltf/java/com/jme3/scene/plugins/gltf/PBRMaterialAdapter.java
+++ b/jme3-plugins/src/gltf/java/com/jme3/scene/plugins/gltf/PBRMaterialAdapter.java
@@ -64,43 +64,34 @@ public abstract class PBRMaterialAdapter extends MaterialAdapter {
 
     @Override
     protected MatParam adaptMatParam(MatParam param) {
-        switch (param.getName()) {
-            case "alpha":
-                String alphaMode = (String) param.getValue();
-                switch (alphaMode) {
-                    case "MASK":
-                        // "MASK" -> BlendMode.Off
-                        getMaterial().setFloat("AlphaDiscardThreshold", MASK_ALPHA_DISCARD);
-                        break;
-                    case "BLEND":
-                        getMaterial().getAdditionalRenderState().setBlendMode(RenderState.BlendMode.Alpha);
-                        break;
-                }
-
-                // Alpha is a RenderState not a Material Parameter, so return null
-                return null;
-            case "doubleSided":
-                boolean doubleSided = (boolean) param.getValue();
-                if (doubleSided) {
-                    //Note that this is not completely right as normals on the back side will be in the wrong direction.
-                    getMaterial().getAdditionalRenderState().setFaceCullMode(RenderState.FaceCullMode.Off);
-                }
-
-                // FaceCulling is a RenderState not a Material Parameter, so return null
-                return null;
-            case "NormalMap":
-                //Set the normal map type to OpenGl
-                getMaterial().setFloat("NormalType", 1.0f);
-                break;
-            case "LightMap":
-                //Gltf only supports AO maps (gray scales and only the r channel must be read)
-                getMaterial().setBoolean("LightMapAsAOMap", true);
-                break;
-            case "alphaCutoff":
-                getMaterial().setFloat("AlphaDiscardThreshold", (float) param.getValue());
-                break;
-            default:
-                break;
+        if (param.getName().equals("alpha")) {
+            String alphaMode = (String) param.getValue();
+            switch (alphaMode) {
+                case "MASK":
+                    // "MASK" -> BlendMode.Off
+                    getMaterial().setFloat("AlphaDiscardThreshold", MASK_ALPHA_DISCARD);
+                    break;
+                case "BLEND":
+                    getMaterial().getAdditionalRenderState().setBlendMode(RenderState.BlendMode.Alpha);
+                    // Alpha is a RenderState not a Material Parameter, so return null
+                    return null;
+            }
+        } else if (param.getName().equals("doubleSided")) {
+            boolean doubleSided = (boolean) param.getValue();
+            if (doubleSided) {
+                //Note that this is not completely right as normals on the back side will be in the wrong direction.
+                getMaterial().getAdditionalRenderState().setFaceCullMode(RenderState.FaceCullMode.Off);
+            }
+            // FaceCulling is a RenderState not a Material Parameter, so return null
+            return null;
+        } else if (param.getName().equals("NormalMap")) {
+            //Set the normal map type to OpenGl
+            getMaterial().setFloat("NormalType", 1.0f);
+        } else if (param.getName().equals("LightMap")) {
+            //Gltf only supports AO maps (gray scales and only the r channel must be read)
+            getMaterial().setBoolean("LightMapAsAOMap", true);
+        } else if (param.getName().equals("alphaCutoff")) {
+            getMaterial().setFloat("AlphaDiscardThreshold", (float)param.getValue());
         }
 
         return param;

